Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -2222,11 +2246,11 @@ private static decimal RoundPrice(decimal input, decimal minTick)
         /// <returns>The new IB ID</returns>
         private int GetNextBrokerageOrderId()
         {
-            // spin until we get a next valid id, this should only execute if we create a new instance
-            // and immediately try to place an order
-            while (_nextValidId == 0) { Thread.Yield(); }
-
-            return Interlocked.Increment(ref _nextValidId);
+            lock (_nextValidIdLocker)
```

Comment:
Could there be a race condition between the call to `reqAllOpenOrders` and setting the new `_nextValidId` and another thread coming through `GetNextBrokerageOrderId()`, in live trading we use a separate thread for processing the orders. Maybe when we call `reqAllOpenOrders` we should already have the lock to `_nextValidIdLocker` so that we can update the id before any new usage

Reply:
I think the race condition between `reqAllOpenOrders` and `GetNextBrokerageOrderId` should never happen because `IBrokerage.GetOpenOrders` is only called once at startup in `BrokerageSetupHandler.Setup`